### PR TITLE
Default value in saveInput

### PR DESCRIPTION
**Pull request by ShrijanL**
- if null=false, default value is given, user value is given, takes user value
- if null=false, default value is given, user value is not given, takes user value

### CODE REVIEW

File: django_generic_api/services.py [hunk 1]
```diff
@@ -164,15 +164,26 @@ def get_model_config_schema(model):
 
         if field_type:
             if is_optional:
```
Comment (jknaresh) [inline django_generic_api/services.py:166]: i think we have to reverse this process, like

is_optional = field1.null or field1.blank ur field1.has_default()

if it has default value, and input empty or null. model will update with default value. or any other idea?

**Review by jknaresh — CHANGES_REQUESTED** [review]
think and optimise

File: django_generic_api/services.py (hunk 1 above)
Comment (ShrijanL) [reply django_generic_api/services.py:171]: takes default value if null=false and input value is set as null.

File: django_generic_api/services.py [hunk 2]
```diff
@@ -132,50 +163,40 @@ def get_model_config_schema(model):
...
-                    field_constraints["max_length"] = field1.max_length
-                break
-
-        if field_type:
-            if is_optional:
-                model_fields[field1.column] = (
-                    field_type[0],  # Optional type
-                    Field(None, **field_constraints),
+        # Retrieve the Pydantic type from the mapping
+        if django_field_name in DJANGO_TO_PYDANTIC_TYPE_MAP:
```
Comment (nareshelequin) [inline django_generic_api/services.py:174]: What if django field not found in map dict?
Use `dict.get` instead `in` while checking.

File: django_generic_api/services.py [hunk 3]
```diff
@@ -132,50 +163,40 @@ def get_model_config_schema(model):
...
                 )
             else:
-                model_fields[field1.column] = (
-                    field_type[0],  # Required type
-                    Field(..., **field_constraints),
+                field_type = (
+                    Optional[mapped_type] if is_optional else mapped_type
                 )
 
+            # Apply constraints such as `max_length` if they exist for the field
```
Comment (nareshelequin) [inline django_generic_api/services.py:191]: Apply constraints such as `max_length` if they exist for the field???
Seems assignment to pydontic model field?

File: django_generic_api/services.py [hunk 4]
```diff
@@ -331,6 +352,22 @@ def handle_save_input(model, record_id, save_input):
                 {"error": f"Extra field {results}", "code": "DGA-S009"}
             )
 
+        for field_name, value in list(saveInput.items()):
```
Comment (nareshelequin) [inline django_generic_api/services.py:355]: For field_name, value in list(saveInput.items()): ??? 

list of dict.items()?

use directly, dict.items()?

File: django_generic_api/services.py [hunk 5]
```diff
@@ -331,6 +352,22 @@ def handle_save_input(model, record_id, save_input):
...
+
+            if (
+                model_field.has_default()
+                and not value
+                and not model_field.null
+            ):
+                saveInput.pop(field_name)
+
+            try:
+                model_field.get_prep_value(value)
```
Comment (nareshelequin) [inline django_generic_api/services.py:367]: FYI, apply same behavior with fetch filter?

**Review by nareshelequin — CHANGES_REQUESTED** [review]

File: django_generic_api/services.py (hunk 2 above)
Comment (ShrijanL) [reply django_generic_api/services.py:174]: raise error if django field not found in map dict,
used .get to get value from dict.

File: django_generic_api/services.py (hunk 3 above)
Comment (ShrijanL) [reply django_generic_api/services.py:191]: Changed comment as assigning attribute 'max length' if they exist for field.

File: django_generic_api/services.py (hunk 5 above)
Comment (ShrijanL) [reply django_generic_api/services.py:367]: Created a ticket for this.
https://github.com/jknaresh/django_generic_api/issues/40

File: django_generic_api/services.py (hunk 4 above)
Comment (ShrijanL) [reply django_generic_api/services.py:355]: as the dictionary changes its size due to default value while iterating over the dict, it throws a runtime error,
so i converted it to list.

**Review by nareshelequin — APPROVED** [review]

**Pull request merged**